codice/ddf
# DDF-5974 Simplifies logout such that we attempt to log out from a single url

#### What does this PR do?
Simplifies logout such that we attempt to log out from a single URL. This will allow us to remove the use of iframes when logging out of external IdPs and appropriately propagate logout errors to the user.

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@bakejeyner 
@garrettfreibott
@SmithJosh 
@stustison

#### How should this be tested?
- [ ] Verify login and logout works for BASIC and PKI auth
    - On logout, you should see a message to close your browser for both auth types
- [ ] Verify login and logout for OIDC auth
    - On logout, you should see a message that you are logged out with a link to sign back in
- [ ] Verify login and logout for SAML auth
    - On logout, you should see the logout page of the SAML provider
- [ ] Verify the logout link on the 403 error page
    - To get the 403 error page, log in to Intrigue using OIDC auth then navigate to the admin page
    - On logout, you should see a message that you are logged out with a link to sign back in. The link should take you to the admin page

#### What are the relevant tickets?
Fixes: #5974

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
